### DISCUSSION TITLE
Third Party Payment Implementation

### DISCUSSION THREAD

**Discussion by Bhanupundhir** [cypress-io/cypress#26464] [Questions and Help]
Hi,

Charge bee is getting displayed in the new tab which does not contain target.

Need quick solution to implement 'Charge bee' third party payment flow through Cypress. Kindly help

**Comment by portyu9** [comment]
Hi @Bhanupundhir 

If Chargebee opens the payment page with `window.open()`, there may be no `target="_blank"` attribute to remove. That is why the usual “remove the target” workaround does not apply in every case.

For most tests, I would avoid driving the provider's second tab. Instead, stub `window.open()` and verify where your application tries to send the user:

```js
cy.window().then((win) => {
  cy.stub(win, 'open').as('windowOpen')
})

cy.get('[data-cy=checkout]').click()

cy.get('@windowOpen').should('have.been.called')
```

You can inspect the URL passed to the stub and, when appropriate, visit that URL in Cypress's existing tab. If you then interact with a different origin, use `cy.origin()` for those commands.

This also keeps the test focused on your application's responsibility: launching the correct Chargebee checkout flow. The payment provider's own UI generally should not need to be exhaustively tested by your Cypress suite.

If exercising the **actual second browser tab** is itself a requirement, current Cypress provides `@cypress/puppeteer` for browser automation scenarios such as multi-tab testing. Normal Cypress command chains still do not operate like a traditional multi-tab WebDriver session.

Please mark this answer as accepted if it helped, thank you.
